docs/spec/v1beta2/providers.md

```diff
@@ -0,0 +1,1406 @@
+# Providers
+
+The `Provider` API defines how events are encoded and where to send them.
+
+## Example
+
+The following is an example of how to send alerts to Slack when Flux fails to
+install or upgrade [Flagger](https://github.com/fluxcd/flagger).
+
+```yaml
+---
+apiVersion: notification.toolkit.fluxcd.io/v1beta2
+kind: Provider
+metadata:
+  name: slack-bot
+  namespace: flagger-system
+spec:
+  type: slack
+  channel: general
+  address: https://slack.com/api/chat.postMessage
+  secretRef:
+    name: slack-bot-token
+---
+apiVersion: notification.toolkit.fluxcd.io/v1beta2
+kind: Alert
+metadata:
+  name: slack
+  namespace: flagger-system
+spec:
+  summary: "Flagger impacted in us-east-2"
+  providerRef:
+    name: slack-bot
+  eventSeverity: error
+  eventSources:
+    - kind: HelmRepository
+      name: '*'
+    - kind: HelmRelease
+      name: '*'
+```
+
+In the above example:
+
+- A Provider named `slack-bot` is created, indicated by the
+  `Provider.metadata.name` field.
+- An Alert named `slack` is created, indicated by the
+  `Alert.metadata.name` field.
+- The Alert references the `slack-bot` provider, indicated by the
+  `Alert.spec.providerRef` field.
+- The notification-controller starts listening for events sent for
+  all HelmRepositories and HelmReleases in the `flagger-system` namespace.
```

Comment:
Would this not listen to cluster wide HelmRepositories and HelmReleases given the default: `--no-cross-namespace-refs=false`?

Reply:
No, it only works for the specified namespace, we don’t support * for namespaces.